narwhals/_expression_parsing.py

```diff
@@ -6,6 +6,7 @@
 from copy import copy
 from typing import TYPE_CHECKING
 from typing import Any
+from typing import Hashable
```

Comment:
Shall we adopt `collections.abc.Hashable`? [typing.Hashable](https://docs.python.org/3/library/typing.html#typing.Hashable) is getting deprecated